### PR TITLE
bug: create/update items convert extent to string frm bbox

### LINKED ISSUE #1023: processParams not converting extent to string
### Describe the bug

`addItem` and `updateItem` both expect `extent` to be a flattened string instead of a bbox ( `1, 2, 3, 4` vs `[[1, 2], [3, 4]]` ). Apparently around two years ago we made a change to `processParams` ( #779 ) for catalog to allow array of arrays through. At that time it was decided to do that for all the cases so as to not make an explicit reference to a property. Unfortunately that has broken `extent`. What ends up happening is in the actual request the bbox is split into two different extent params, the create/update call succeeds, however the item ends up with an extent which is simply an empty array.

### Reproduction

Will be provided

### Logs

_No response_

### System Info

```shell
v3.4.3
```

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by benstoltz**
Resolves #1023

### CODE REVIEW

File: packages/arcgis-rest-portal/src/items/create.ts [hunk 1]
```diff
@@ -98,6 +100,11 @@ export function createItemInFolder(
       ...serializeItem(requestOptions.item)
     };
 
+    // convert extent, if present, into a string from bbox
```
Comment (dbouwman) [inline packages/arcgis-rest-portal/src/items/create.ts:103]: I'd have done this a little higher so it's clear we're working with the `item.extent`... this is functionally correct though

File: packages/arcgis-rest-portal/src/items/helpers.ts [hunk 2]
```diff
@@ -290,3 +290,26 @@ export function determineOwner(requestOptions: any): Promise<string> {
...
+};
+
+/**
+ * Given a Bbox, convert it to a string. Some api endpoints expect a string
+ *
+ * @param {BBox} extent
+ * @return {*}  {string}
+ */
+ export function bboxToString (extent: number[][]): string {
+  return extent.map((a) => a.join(", ")).join(", ");
```
Comment (dbouwman) [inline packages/arcgis-rest-portal/src/items/helpers.ts:314]: while I'm sure this works, it seems that `[[1,2],[3,4]].join(',')` returns `1,2,3,4` and I think that's the old implementation in process params.

**Review by dbouwman — APPROVED** [review]
This is good with me - maybe add a comment where we're doing `.params.extent` just to inform the next dev what's going on.

File: packages/arcgis-rest-portal/src/items/create.ts (hunk 1 above)
Comment (benstoltz) [reply packages/arcgis-rest-portal/src/items/create.ts:103]: Problem with that is doing it higher means manipulating an IItemAdd, which expects that extent is a `number[][]` which means doing the whole `as unknown as number[][]`.

File: packages/arcgis-rest-portal/src/items/helpers.ts (hunk 2 above)
Comment (benstoltz) [reply packages/arcgis-rest-portal/src/items/helpers.ts:314]: Ah, I was going off the docs which show the output like: `1, 2, 3, 4` (`extent=<xmin>, <ymin>, <xmax>, <ymax>` ; see [addItem](https://developers.arcgis.com/rest/users-groups-and-items/add-item.htm))

**Review by patrickarlt — APPROVED** [review]
LGTM!

**Pull request merged**